duecredit/io.py

```diff
@@ -125,8 +128,9 @@ def _filter_citations(self, tags=None):
         cited_modobj = list(modules) + list(objects)
         for package in cited_packages:
             package_citations = packages[package]
-            if list(filter(lambda x: x.cite_module, package_citations)) or \
-               list(filter(lambda x: _is_contained(package, x), cited_modobj)):
+            if all_ or \
+                list(filter(lambda x: x.cite_module, package_citations)) or \
+                list(filter(lambda x: _is_contained(package, x), cited_modobj)):
                 continue
```

Comment:
Above I think you want to use `any` instead of `list`
Unlikely to be of noteable benefit but it would be faster, terminating as soon as any entry is found